### PR TITLE
102 Change ensemble training

### LINKED ISSUE #102: Fix reward ensemble training
### The problem
- Every model in the ensemble trains on the same data, this resulting in low varianz between them.
### The solution / the feature
- This will enhance the reward model

### PR DESCRIPTION
**Pull request by Wizzardexe**
**Issue:** Fix #102

**What?**
- Change reward ensemble training

**How?**
- I moved the sampling of the trainings buffer in side of the training of each ensemble

**Ready to merge to main?**
- [x] Linter passes
- [x] PR Approved

### CODE REVIEW

File: framework/reward_net.py [hunk 1]
```diff
@@ -343,36 +340,24 @@ def train_or_val_pref_batch(
...
-                    )
-                    pred_u = model.preference_prob(r1_u, r2_u)
-                    loss_u = model.preference_loss(pred_u, pseudo_label)
-                    assert loss_u != float("inf")
-                    ensemble_loss_u = ensemble_loss_u + loss_u
-
-                ensemble_loss_u /= len(model.ensemble)
-                batch_loss = batch_loss + ensemble_loss_u
-                unsup_loss_accum += ensemble_loss_u.item()
+                batch_loss = batch_loss + loss_u
```
Comment (flyingpalli) [inline framework/reward_net.py:359]: this can be `+=` if I'm not mistaken

**Review by flyingpalli — APPROVED** [review]
These changes seem sufficient to implement the discussed behavior

File: framework/reward_net.py (hunk 1 above)
Comment (Wizzardexe) [reply framework/reward_net.py:359]: You can't use this shortcut with a tensor, which needs grad.

**Review by bkuen — APPROVED** [review]
Yeah, I guess thats what he meant
